### DISCUSSION TITLE
Issue with OTG port on WLAN Pi Pro

### DISCUSSION THREAD

**Discussion by mithun294** [WLAN-Pi/feedback#27] [Hardware - WLAN Pi Pro]
Hi All,
 
I bought WLAN Pi Pro couple of days back.
But I’m having issues with the OTG USB, seems like My MacBook Pro cannot recognize after connecting to the USB OTG port.
I cannot access the Web GUI using the Link Local IP Address 169.254.42.1 (cannot ping as well).
Can you please help me with this? Should I enable the OTG Port or something?

**Comment by jolla** [comment]
OTG is enabled by default. Which model MacBook do you have? We have seen issues with OTG not working on M series MacBooks. If that is the case you might want to try connecting the WLAN Pi through a USB hub as some users have found found this works around the issues with M series MacBooks.

**Reply by mithun294** [reply]
I have MacBook M1 with Monterey MacOS. Well the Hub worked but its not charging using OTG port. IS that normal?

**Reply by jolla** [reply]
Good to hear that worked at least as a work around until we can get to the bottom of it, it should still charge as well.

**Comment by mithun294** [comment]
But It says battery is discharging and the connection is not very stable.

**Comment by jolla** [comment]
If it shows it is discharging that would indicate it is not receiving enough power, this could be a limitation of your USB cable or USB hub. Do you have a different USB cable or USB hub/adapter you could test?

**Reply by mithun294** [reply]
I tried with the multiple USB HUBs and cables. Doesn't seem to be working.
Need to get OTG port working somehow. Please help

**Reply by jolla** [reply]
The USB Type-C port on the top of the WLAN Pi Pro is both charging port and a data port. Could you provide a bit more detail on what isn't working? If you connect the top USB Type-C port to your laptops USB port you should see a USB RNDIS connection in your network list. The WLAN Pi should be reachable with the IP address of 169.254.42.1.

**Comment by mithun294** [comment]
By not working I mean I cannot see USB RNDIS connection on my MacBook Pro.
So I'm unable to reach 169.254.42.1

**Reply by jolla** [reply]
Unfortunately I don't have an M1 Mac to test with, but my understanding is there is an issue with the M series Macs and OTG, a work around that has worked for other M1 Mac users is to connect the WLAN Pi Pro through a USB-C hub. 

Perhaps @adriangranados @nickjvturner can share more info on this as they have access to M series Macs.

**Reply by adriangranados** [reply]
OTG over USB-C doesn't work in the M1 (this is an M1 Mac issue). The workaround is to use a USB-C hub with USB-A ports, and have the WLAN Pi Pro connect to the hub using a USB-C (top port in the WLAN Pi Pro) to USB-A cable/adapter, then connect the hub to the Mac using USB-C.

**Reply by mithun294** [reply]
I tried using multiple USB Hubs but still M1 doesn't seem to recognize OTG Port.
Can you please tell me the which model of USB hub will work for sure?

**Reply by adriangranados** [reply]
I'm using this hub: https://www.amazon.com/Hiearcool-USB-Hub-11-Compatiable/dp/B07QNRM45T/ref=sr_1_4?crid=297E3KBAF0XBZ&keywords=ucn3270&qid=1663945095&sprefix=ucn3270%2Caps%2C105&sr=8-4&ufe=app_do%3Aamzn1.fos.08f69ac3-fd3d-4b88-bca2-8997e41410bb

I'm also using the USB-C cable that came with the WLAN Pi Pro, and a USB-C to USB-A adapter. Then, I connect the Pro to the hub on one of the USB 3.0 ports (although it works on the 2.0 ports too I think).
